beacon/message.go

```diff
@@ -17,31 +17,35 @@ type ProposalVrfMessage struct {
 type ProposalMessage struct {
 	EpochID      types.EpochID
 	NodeID       types.NodeID
-	VRFSignature []byte
+	VRFSignature []byte `scale:"max=32"`
+}
+
+type Proposal struct {
+	Value []byte `scale:"max=32"` // TODO(mafa): check if max. value is correct
 }
 
 // FirstVotingMessageBody is FirstVotingMessage without a signature.
 type FirstVotingMessageBody struct {
 	EpochID                   types.EpochID
-	ValidProposals            [][]byte
-	PotentiallyValidProposals [][]byte
+	ValidProposals            []Proposal `scale:"max=32"` // TODO(mafa): check if max. value is correct
+	PotentiallyValidProposals []Proposal `scale:"max=32"` // TODO(mafa): check if max. value is correct
```

Comment:
maybe it makes sense to change Proposal to byte array (e.g [32]byte ) ? it is more compact, easier to decode and safer

@countvonzero

Reply:
In some places (like signatures) this might work, but the size of `ValidProposals` and `PotentiallyValidProposals` is afaik a configuration parameter.

Arrays cannot be dynamic in size, that being said the `scale:"max=xxx"` tag also doesn't consider configuration...

Reply:
yes i was talking about that

we need to set some upper bound on the total number of proposals as well, maybe 10000 is ok ?

Reply:
here the total number of proposals, with high probabilities, would be lower than 100. so i think 200 is enough.

Reply:
Made Proposal a `[4]byte` instead of `[]byte` and added a max of 200 to `ValidProposals` and `PotentiallyValidProposals`